File: JavaScript with Axios/unrestricted-pdf-endpoint.js
/**
 * Use this request to remove security restrictions from a password protected PDF.
 * NOTE: By default, removing the permissions password will also remove encryption by open password. To keep an open password on a document, include the current_open_password form-data field set to the correct open password.
 */
var axios = require('axios');
var FormData = require('form-data');
var fs = require('fs');

// Create a new form data instance and append the PDF file and parameters to it
var data = new FormData();
data.append('file', fs.createReadStream('/path/to/file'));
data.append('current_permissions_password', 'current_example_pw');
data.append('output', 'pdfrest_unrestricted_pdf');

// define configuration options for axios request
var config = {
  method: 'post',
  maxBodyLength: Infinity, // set maximum length of the request body
  url: 'https://api.pdfrest.com/unrestricted-pdf', 
  headers: { 
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', // Replace with your API key
    ...data.getHeaders() // set headers for the request
  },
  data : data // set the data to be sent with the request
};

// send request and handle response or error
axios(config)
.then(function (response) {
  console.log(JSON.stringify(response.data));
})
.catch(function (error) {
  console.log(error); 
});

// If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.js' sample.

File: Python/unrestricted-pdf-endpoint.py
import requests

# Set the API endpoint URL
url = "https://api.pdfrest.com/unrestricted-pdf"

# Define the payload data for the request
payload = {
    'current_permissions_password': 'current_example_pw',
    'output': 'pdfrest_unrestricted_pdf'
}

# Specify the PDF file to remove restrictions from
files = [
    ('file', ('file', open('/path/to/file', 'rb'), 'application/octet-stream'))
]

# Set the headers, including the API key
headers = {
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'  # Replace with your API key
}

# Make a POST request to the API endpoint
response = requests.request(
    "POST", url, headers=headers, data=payload, files=files)

# Print the response content
print(response.text)
